userspace/libsinsp/threadinfo.cpp

```diff
@@ -899,23 +899,14 @@ std::string sinsp_threadinfo::get_cwd()
 
 void sinsp_threadinfo::set_cwd(const char* cwd, uint32_t cwdlen)
 {
-	char tpath[SCAP_MAX_PATH_SIZE];
 	sinsp_threadinfo* tinfo = get_main_thread();
 
 	if(tinfo)
 	{
-		sinsp_utils::concatenate_paths(tpath,
-			SCAP_MAX_PATH_SIZE,
-			(char*)tinfo->m_cwd.c_str(),
-			(uint32_t)tinfo->m_cwd.size(),
-			cwd,
-			cwdlen);
+		tinfo->m_cwd = sinsp_utils::concatenate_paths(m_cwd, std::string_view(cwd, cwdlen)).c_str();
+		uint32_t len = tinfo->m_cwd.length();
 
-		tinfo->m_cwd = tpath;
-
-		uint32_t size = tinfo->m_cwd.size();
-
-		if(size == 0 || (tinfo->m_cwd[size - 1] != '/'))
+		if(len == 0 || (tinfo->m_cwd[len - 1] != '/'))
```

Comment:
No need for `len`.
```suggestion
		if(tinfo->m_cwd.empty() || tinfo->m_cwd.back() != '/'))
```

Reply:
That's better, after you respond to above's comment I can rewind the last 2 commits.